rstar/Cargo.toml

```diff
@@ -16,14 +16,16 @@ maintenance = { status = "actively-developed" }
 
 [dependencies]
 heapless = "0.6"
-num-traits = "0.2"
+num-traits = { version = "0.2", default-features = false }
 pdqselect = "=0.1.0"
-serde = { version = "1.0", optional = true, features = ["derive"] }
+serde = { version = "1.0", optional = true, default-features = false, features = ["derive"] }
 smallvec = "1.6"
 
 [features]
-default = []
+default = ["std"]
 debug = []
+std = [ "num-traits/default" ]
+alloc = [ "num-traits/libm" ]
```

Comment:
Is there a specific need for the `alloc` feature? Users might forget to include it after choosing `--no-default-features`.

We could, instead, replace any `cfg(feature = alloc)` with `cfg(not(feature = "std"))` and _always_ use `num_traits` with the `libm` feature.

However, I'm okay with leaving this as is if there's some reason for it that I'm not aware of. Working with no-std is fairly new to me, please excuse if I'm not aware of conventions and best practices in this area.

Reply:
I believe omitting the `alloc` should be possible, exactly the way you proposed. The question here is: can the `rstar` crate deliver any value as library if all things which depend on `alloc` are removed? If not, I'm happy to factor out the `alloc` feature just as you described.

Reply:
From my understanding, not having `alloc` would require to "reserve" a fixed amount of memory on the stack and using that for representing the tree.
This would require to represent the tree as part of a continuous segment of memory - it's currently modelled as a "real" tree on the heap (See struct `ParentNodeData`).
While that is something that could be interesting (e.g. just for the performance benefits it may bring), it's also a lot of work and restructuring that will be required. Thus, I wouldn't bet on this feature (unless we find someone wanting to work on this) and would be in **favour of leaving out `alloc` as a feature**.

Sorry for the lengthy explanation - I'm giving it in the hope to be corrected in case anything is represented inaccurately.

Reply:
Ok, then there is some things to discuss left: AFAIK there is no negative feature lookup, e. g. when there is no specific feature for `no_std` support, we can not conditionally depend on `alloc` in the Cargo.toml. However, we can also get away without using features at all! We can just always depend on `alloc` and `core`, and the lib is unconditionally always `no_std`. This should not affect downstream consumers, everybody who either

+ offers an `alloc` implementation
+ compiles with std

can use the rstar. Are you fine if we pursue this way?

Reply:
Good observation! I didn't look at it from this angle.
Let's go for the feature-less approach that you have described.

Reply:
Ok, I updated the commit accordingly. Please revisit your TODO items, they probably changed now?